hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -56,8 +57,11 @@ def __init__(self,
         self.train()
 
         if pretrained_model_path is not None:
-            assert isinstance(pretrained_model_path, str), f"Expected a string, got {type(pretrained_model_path)}"
-            self.load_state_dict(torch.load(pretrained_model_path))
+            if not isinstance(pretrained_model_path, (str, Path)):
+                raise TypeError(f"Expected a string or Path, got {type(pretrained_model_path)}")
+            device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
I think we shouldn't make such assumptions. The functions role is just to load the weights, it's up to the application and outer context to place it in appropriate compute device (cuda:0, cuda:1, etc.)

Reply:
The problem is that the weights were saved on GPU, so this fails in a machine without one. I agree it would be up to the application to place it in an appropriate device for inference. Shall we default to always load to `"cpu"`?

Reply:
I can update the weigths, give me 10 mins. We can push a change.

Reply:
Good idea -- discussed it offline.

Reply:
In https://github.com/microsoft/hi-ml/pull/497/commits/30fb7ff62a05b9de5275ff76802c7f7effc5df33:

- Use `"cpu"` as map location when loading weights
- Add `to` method to joint inference engine and `to` attribute to the image and text engines
- Automatically select the device in the notebook and call the new method
- In the inference engines, choose device from model at inference time, not at instantiation time